deltacat/compute/compactor_v2/utils/compaction_utils.py

```diff
@@ -0,0 +1,800 @@
+    )
+
+    base_audit_url = rcf_source_partition_locator.path(
+        f"s3://{params.compaction_artifact_s3_bucket}/compaction-audit"
+    )
+    audit_url = f"{base_audit_url}.json"
+    logger.info(f"Compaction audit will be written to {audit_url}")
+    mutable_compaction_audit = (
+        CompactionSessionAuditInfo(deltacat.__version__, ray.__version__, audit_url)
+        .set_hash_bucket_count(params.hash_bucket_count)
+        .set_compactor_version(CompactorVersion.V2.value)
+    )
+
+    compaction_start = time.monotonic()
+
+    task_max_parallelism: int = params.task_max_parallelism
+
+    if params.pg_config:
+        logger.info(
+            "pg_config specified. Tasks will be scheduled in a placement group."
+        )
+        cluster_resources = params.pg_config.resource
+        cluster_cpus = cluster_resources["CPU"]
+        cluster_memory = cluster_resources["memory"]
+        task_max_parallelism = cluster_cpus
+        mutable_compaction_audit.set_total_cluster_memory_bytes(cluster_memory)
+
+    # read the results from any previously completed compaction round
+    round_completion_info: Optional[RoundCompletionInfo] = None
+    high_watermark: Optional[HighWatermark] = None
+    previous_compacted_delta_manifest: Optional[Manifest] = None
+
+    if not params.rebase_source_partition_locator:
+        round_completion_info = rcf.read_round_completion_file(
+            params.compaction_artifact_s3_bucket,
+            params.source_partition_locator,
+            params.destination_partition_locator,
+            **params.s3_client_kwargs,
+        )
+        if not round_completion_info:
+            logger.info(
+                "Both rebase partition and round completion file not found. Performing an entire backfill on source."
+            )
+        else:
+            compacted_delta_locator = round_completion_info.compacted_delta_locator
+
+            previous_compacted_delta_manifest = (
+                params.deltacat_storage.get_delta_manifest(
+                    compacted_delta_locator, **params.deltacat_storage_kwargs
+                )
+            )
+
+            high_watermark = round_completion_info.high_watermark
+            logger.info(f"Setting round completion high watermark: {high_watermark}")
+            assert (
+                params.hash_bucket_count == round_completion_info.hash_bucket_count
+            ), (
+                "The hash bucket count has changed. "
+                "Kindly run rebase compaction and trigger incremental again. "
+                f"Hash Bucket count in RCF={round_completion_info.hash_bucket_count} "
+                f"not equal to Hash bucket count in args={params.hash_bucket_count}."
+            )
+
+        logger.info(f"Round completion file: {round_completion_info}")
+    return (
+        mutable_compaction_audit,
+        previous_compacted_delta_manifest,
+        round_completion_info,
+        audit_url,
```

Comment:
Is the `audit_url` already present in `mutable_compaction_audit`?
If so, might not be necessary to return `audit_url` as a separate parameter.